src/TwigComponent/src/Twig/ComponentLexer.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Symfony\UX\TwigComponent\Twig;
+
+use Twig\Lexer;
+use Twig\Source;
+use Twig\TokenStream;
+
+class ComponentLexer extends Lexer
+{
+    public const ATTRIBUTES_REGEX = '(?<attributes>(?:\s+[\w\-:.@]+(=(?:\\\"[^\\\"]*\\\"|\'[^\']*\'|[^\'\\\"=<>]+))?)*\s*)';
+    public const OPEN_TAGS_REGEX = '/<\s*x-(?<name>([[\w\-\:\.]+))\s*'.self::ATTRIBUTES_REGEX.'(\s?)+>/';
```

Comment:
If possible, it'd be great to identify a matching opening tag - e.g. `<x-successAlert` FIRST with the regex. And then walk through the rest of the source looking for attributes and eventually the closing `>` or `/>`. That may be beyond the scope of this PR (and Twig internals are COMPLEX). But the motivation is to give better errors. For example, suppose a user does this:

```
<x-successAlert type='danger>
```

I'm not sure what error that would give currently (it might not even match the regex and then would fail silently - which is kind of the worst situation). But ideally, it would be something like:

> Unexpected character `>`: was expecting `'` to close the attribute string.

Thanks :)